src/io_source.rs

```diff
@@ -0,0 +1,198 @@
+use std::ops::{Deref, DerefMut};
+#[cfg(unix)]
+use std::os::unix::io::AsRawFd;
+#[cfg(windows)]
+use std::os::windows::io::AsRawSocket;
+use std::{fmt, io};
+
+#[cfg(unix)]
+use crate::poll;
+#[cfg(debug_assertions)]
+use crate::poll::SelectorId;
+use crate::sys::IoSourceState;
+use crate::{event, Interest, Registry, Token};
+
+/// Adapter for a [`RawFd`] or [`RawSocket`] providing an [`event::Source`]
+/// implementation.
+///
+/// `IoSource` enables registering any type with an FD or socket with [`Poll`].
```

Comment:
Suggestion: "`IoSource` enables registering any FD or socket wrapper with [`Poll`]"

I suggest this because the current wording of "with an FD or socket" may imply it could be a type with multiple fields; one of which is an FD or socket.